Explain SVG presentation attributes (#44434)
mdn/content: files/en-us/web/svg/reference/attribute/index.md

### Presentation attributes

SVG presentation attributes are SVG attributes that can also be used as CSS properties on SVG elements.

> They set CSS property values on an element and are applied as author-level style rules with specificity `0`, so other author styles in a stylesheet or {{SVGAttr("style")}} attribute can override them.

Reviewer: ```suggestion
They set CSS property values on an element with a specificity of `0`, so other author styles in a stylesheet or {{SVGAttr("style")}} attribute can override them.
```

Presentation attribute values are parsed as CSS values, not declarations, so they cannot contain `!important`.

> [!NOTE]
> Whether these attributes are presentation attributes depends on the element on which they are set. For example, `x` is a presentation attribute for {{svgelement("circle")}}, but not for {{svgelement("tspan")}}; it's the coordinate of the starting point of the text baseline, or the x coordinate of each individual glyph if a list of values is provided.

> #### Geometry properties

Reviewer: The geometry properties subsection should be put after the big list of Presentation attributes, not before. In the current state, it is messy and confusing.

Geometry properties describe the position and dimensions of some SVG elements.

> They are a subset of presentation attributes, but their attribute form is only a presentation attribute on the elements for which the matching CSS property applies.

Reviewer: I'm not an SVG expert, but, isn't this true for _all_ presentation attributes? The points made in the original issue seem to be more like:

- They are only applicable to certain elements.
- Their CSS equivalents do not inherit.

This makes sense, as something like `r` is definitely not relevant to non-circle defining elements, and its CSS property value equivalent really shouldn't be inherited either.

> For example, {{SVGAttr("x")}} is a geometry property for {{SVGElement("rect")}}, {{SVGElement("image")}}, and {{SVGElement("svg")}}, but not for {{SVGElement("text")}} or {{SVGElement("tspan")}}.

Reviewer: This is wrong. If you read the `x` page, you'll see that it applies to `<text>` and `<tspan>`. Is this an AI hallucination that you've not bothered to check?

`r` would be a much better example to include here.

Geometry properties are not inherited.
Setting a geometry property on a container element, such as {{SVGElement("g")}}, has no effect on that element's children.

> SVG geometry properties include {{cssxref("cx")}}, {{cssxref("cy")}}, {{cssxref("r")}}, {{cssxref("rx")}}, {{cssxref("ry")}}, {{cssxref("x")}}, {{cssxref("y")}}, {{cssxref("width")}}, {{cssxref("height")}}, and {{cssxref("d")}}.

Reviewer: You should provide a full list of the attributes that are Geometry properties, not just give a few examples ("SVG geometry properties include..." implies that this is not an exhaustive list).
